cypress/e2e/awx/resources/hosts.cy.ts

```diff
@@ -229,6 +193,72 @@ describe('host and inventory host', () => {
+    // boolean value will be false for stand alone host tests
+    cy.createInventoryHostGroup(organization).then((result) => {
+      const { inventory, host, group } = result;
+      let url = '';
+      if (inventory_host) {
+        url = `/infrastructure/inventories/inventory/${inventory.id}/hosts/?page=1&perPage=10&sort=name`;
+      } else {
+        url = '/infrastructure/hosts?page=1&perPage=10&sort=name';
+      }
+      // TODO: unsafe assignment, find a better way to retrieve host id it doesn't return always
+      const hostid = host.id ? host.id.toString() : '';
+      navigateToHost(url, host.name, '[data-cy="name-column-cell"] a');
+      expect(host.inventory).to.eq(inventory.id);
+      expect(group.inventory).to.eq(inventory.id);
+      cy.clickLink(/^Groups$/);
+      //check edit group
+      cy.getByDataCy('edit-group').click();
+      cy.verifyPageTitle('Edit group');
+      cy.getByDataCy('name-form-group').type('-changed');
+      cy.getByDataCy('Submit').click();
+      cy.verifyPageTitle(group.name + '-changed');
+      cy.requestPost<{ name: string; inventory: number; id: number }>(
+        awxAPI`/hosts/${hostid}/groups/`,
+        {
+          name: 'E2E Group ' + randomString(5),
+          inventory: host.inventory,
+        }
+      ).then((group: { name: string; id: number }) => {
+        /// check multiple associate and disassociate
+        // disassociate
+        navigateToHost(url, host.name, '[data-cy="name-column-cell"] a');
+        cy.clickLink(/^Groups$/);
+        cy.getByDataCy('select-all').click();
+        disassociate();
+        cy.getByDataCy('empty-state-title').contains(
+          /^There are currently no groups associated with this host/
+        );
+        // Add - multi groups
+        cy.clickButton(/^Associate groups$/);
+        cy.getByDataCy('select-all').click();
+        cy.clickModalButton('Confirm');
+        cy.contains('button', 'Close').click();
+        cy.contains(group.name);
+        /// single disassociate
+        // TODO: need to change this when
+        // https://issues.redhat.com/browse/AAP-22914 change will applyed
+        cy.filterTableByMultiSelect('name', [group.name]);
+        cy.get(`[data-cy="row-id-${group.id}"] [data-cy="checkbox-column-cell"]`).click();
+        disassociate();
+        navigateToHost(url, host.name, '[data-cy="name-column-cell"] a');
+        cy.clickLink(/^Groups$/);
+        cy.contains(group.name).should('not.exist');
+        //check single associate
+        cy.getByDataCy('associate').click();
+        cy.get(`[data-cy="row-id-${group.id}"] [data-cy="checkbox-column-cell"]`).click();
+        cy.clickModalButton('Confirm');
+        cy.contains('button', 'Close').click();
+        cy.contains(group.name);
+        deleteHost(inventory.id, inventory_host, host.name);
+      });
+    });
+  }
+
   function createAndEditAndDeleteHost(inventory_host: boolean, inventory: Inventory) {
```

Comment:
I will suggest maybe different parameter, such as type: 'standalone_host' | 'inventory_host', because this way,  the call of the function is more readable, such as:

createAndEditAndDeleteHost('standalone_host')

this way it is much more clear what is that parameter call, true/false is not clear.

Reply:
do you mean pass a string?
as for now I've added a lot of comments to try and make it clearer 
I can add for this as well

Reply:
changed it base on suggestion - now it's string